app/src/test/java/org/oppia/app/topic/questionplayer/QuestionPlayerActivityLocalTest.kt

```diff
@@ -0,0 +1,287 @@
+  }
+
+  @Test
+  fun testQuestionPlayer_submitTwoWrongAnswers_expandResponse_checkPreviousHeaderExpanded() {
+    launchForQuestionPlayer(SKILL_ID_LIST).use {
+      testCoroutineDispatchers.advanceTimeBy(TimeUnit.SECONDS.toMillis(60))
+      onView(withId(R.id.question_recycler_view)).check(matches(isDisplayed()))
+
+      submitTwoWrongAnswersToQuestionPlayer()
+
+      onView(withId(R.id.previous_response_header)).check(matches(isDisplayed()))
+      onView(withId(R.id.question_recycler_view))
+        .perform(scrollToViewType(StateItemViewModel.ViewType.PREVIOUS_RESPONSES_HEADER))
+      onView(withId(R.id.previous_response_header)).perform(ViewActions.click())
+      onView(withId(R.id.question_recycler_view)).check(
+        matches(
+          hasChildCount(/* childCount= */6)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testQuestionPlayer_expandCollapseResponse_checkPreviousHeaderCollapsed() {
+    launchForQuestionPlayer(SKILL_ID_LIST).use {
+      testCoroutineDispatchers.advanceTimeBy(TimeUnit.SECONDS.toMillis(60))
+      onView(withId(R.id.question_recycler_view)).check(matches(isDisplayed()))
+
+      submitTwoWrongAnswersToQuestionPlayer()
+
+      onView(withId(R.id.previous_response_header)).check(matches(isDisplayed()))
+      onView(withId(R.id.question_recycler_view)).check(
+        matches(
+          hasChildCount(/* childCount= */5)
+        )
+      )
+
+      onView(withId(R.id.question_recycler_view))
+        .perform(scrollToViewType(StateItemViewModel.ViewType.PREVIOUS_RESPONSES_HEADER))
+      onView(withId(R.id.previous_response_header)).perform(ViewActions.click())
+      onView(withId(R.id.question_recycler_view)).check(
+        matches(
+          hasChildCount(/* childCount= */6)
+        )
+      )
+
+      onView(withId(R.id.previous_response_header)).perform(ViewActions.click())
+      onView(withId(R.id.question_recycler_view)).check(
+        matches(
+          hasChildCount(/* childCount= */5)
+        )
+      )
+    }
+  }
+
+  private fun launchForQuestionPlayer(
+    skillIdList: ArrayList<String>
+  ): ActivityScenario<QuestionPlayerActivity> {
+    return ActivityScenario.launch(
+      QuestionPlayerActivity.createQuestionPlayerActivityIntent(
+        context, skillIdList
+      )
+    )
+  }
+
+  private fun submitTwoWrongAnswersToQuestionPlayer() {
+    submitWrongAnswerToQuestionPlayerFractionInput()
+    submitWrongAnswerToQuestionPlayerFractionInput()
+  }
+
+  private fun submitWrongAnswerToQuestionPlayerFractionInput() {
+    onView(withId(R.id.question_recycler_view))
+      .perform(scrollToViewType(StateItemViewModel.ViewType.TEXT_INPUT_INTERACTION))
+    onView(withId(R.id.text_input_interaction_view)).perform(appendText("1"))
+    testCoroutineDispatchers.runCurrent()
+
+    onView(withId(R.id.question_recycler_view))
+      .perform(scrollToViewType(StateItemViewModel.ViewType.SUBMIT_ANSWER_BUTTON))
+    onView(withId(R.id.submit_answer_button)).perform(ViewActions.click())
```

Comment:
Nit: prefer to directly import view actions like ``click()``.

Ditto elsewhere.

Reply:
Done